core/src/main/scala/kafka/server/Kip500Broker.scala

```diff
@@ -111,7 +111,10 @@ class Kip500Broker(val config: KafkaConfig,
   var quotaManagers: QuotaFactory.QuotaManagers = null
 
   val brokerMetaPropsFile = "meta.properties"
-  val brokerMetadataCheckpoints = config.logDirs.map(logDir => (logDir, new BrokerMetadataCheckpoint(new File(logDir + File.separator + brokerMetaPropsFile)))).toMap
+
+  // Look for metadata checkpoints (meta.properties) in all log.dirs and metadata.log.dir
+  val brokerMetadataLogDirs = config.logDirs:+(config.metadataLogDir)
+  val brokerMetadataCheckpoints = brokerMetadataLogDirs.map(logDir => (logDir, new BrokerMetadataCheckpoint(new File(logDir + File.separator + brokerMetaPropsFile)))).toMap
```

Comment:
metadata checkpoints appear in all directories, not just log directories

Reply:
Hmm. I just appended `config.metadataLogDir` to the existing list of directories (which are just `config.logDirs`). Are you saying there's more to add here? I'm not sure what 'all directories' here means.

Reply:
Ah. It's the name of the variable I see. Right, I'll fix that.